src/main/java/de/tum/in/www1/artemis/service/exam/ExamImportService.java

```diff
@@ -138,34 +134,73 @@ public List<ExerciseGroup> importExerciseGroupsWithExercisesToExistingExam(List<
-        AtomicInteger numberOfInvalidProgrammingExercises = new AtomicInteger(0);
-        // Iterate over all exercises
-        exerciseGroupList.forEach(exerciseGroup -> {
-            exerciseGroup.getExercises().forEach(exercise -> {
+    private void preCheckProgrammingExercisesForTitleAndShortNameUniqueness(List<ExerciseGroup> exerciseGroups, String courseShortName) {
+        // check for duplicated titles
+        List<String> titles = exerciseGroups.stream().flatMap(group -> group.getExercises().stream()).filter(ex -> ex instanceof ProgrammingExercise).map(BaseExercise::getTitle)
+                .toList();
+        Set<String> uniqueTitles = new HashSet<>(titles);
+
+        if (titles.size() != uniqueTitles.size()) {
+            exerciseGroups.forEach(exerciseGroup -> exerciseGroup.getExercises().forEach(exercise -> {
                 if (exercise.getExerciseType() == ExerciseType.PROGRAMMING) {
-                    // Method to check, if the project already exists.
-                    boolean invalidShortName = programmingExerciseService.preCheckProjectExistsOnVCSOrCI((ProgrammingExercise) exercise, targetCourseShortName);
-                    if (invalidShortName) {
-                        // If the project already exists and thus the short name isn't valid, it is removed
-                        exercise.setShortName("");
+                    if (!uniqueTitles.contains(exercise.getTitle())) {
                         exercise.setTitle("");
-                        numberOfInvalidProgrammingExercises.getAndIncrement();
+                    }
+                    else {
+                        uniqueTitles.remove(exercise.getTitle());
                     }
                 }
-            });
+            }));
+            throw new ExamConfigurationException(exerciseGroups, 0, "duplicatedProgrammingExerciseTitle");
+        }
+
+        // check for duplicated short names
+        List<String> shortNames = exerciseGroups.stream().flatMap(group -> group.getExercises().stream()).filter(ex -> ex instanceof ProgrammingExercise)
+                .map(BaseExercise::getShortName).toList();
+        Set<String> uniqueShortNames = new HashSet<>(shortNames);
+
+        if (shortNames.size() != uniqueShortNames.size()) {
+            exerciseGroups.forEach(exerciseGroup -> exerciseGroup.getExercises().forEach(exercise -> {
+                if (exercise.getExerciseType() == ExerciseType.PROGRAMMING) {
+                    if (!uniqueShortNames.contains(exercise.getShortName())) {
+                        exercise.setShortName("");
+                    }
+                    else {
+                        uniqueShortNames.remove(exercise.getShortName());
+                    }
+                }
+            }));
+            throw new ExamConfigurationException(exerciseGroups, 0, "duplicatedProgrammingExerciseShortName");
+        }
+
+        // Flag to determine, if a programming exercise with an invalid shortName was found
```

Comment:
There are some already existing tests that test the behavior in case the project already exists on VCS/CI. I didn't really change anything there. I moved the forEach into a seperate method :+1: